frameworks/cassandra/src/main/java/com/mesosphere/sdk/cassandra/scheduler/CassandraRecoveryPlanOverrider.java

```diff
@@ -0,0 +1,118 @@
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * The CassandraRecoveryPlanManager handles failure scenarios unique to Cassandra. It falls back to the default recovery
+ * behavior when appropriate.
+ */
+public class CassandraRecoveryPlanOverrider implements RecoveryPlanOverrider {
+    protected final Logger logger = LoggerFactory.getLogger(getClass());
+    private static final String RECOVERY_PHASE_NAME = "permanent-node-failure-recovery";
+    private final StateStore stateStore;
+    private final ConfigStore<ServiceSpec> configStore;
+    private final Plan replaceNodePlan;
+
+    public CassandraRecoveryPlanOverrider(
+            StateStore stateStore,
+            ConfigStore<ServiceSpec> configStore,
+            Plan replaceNodePlan) {
+        this.stateStore = stateStore;
+        this.configStore = configStore;
+        this.replaceNodePlan = replaceNodePlan;
+    }
+
+    @Override
+    public Optional<Phase> override(PodInstanceRequirement stoppedPod) {
+        if (!stoppedPod.getPodInstance().getPod().getType().equals("node")
+                || stoppedPod.getRecoveryType() != RecoveryType.PERMANENT) {
+            logger.info("No overrides necessary. Pod is not a node or it isn't a permanent failure.");
+            return Optional.empty();
+        }
+
+        Phase nnPhase = null;
+        int index = stoppedPod.getPodInstance().getIndex();
+        logger.info(String.format("Returning replacement plan for node %d.", index));
+        return Optional.ofNullable(getNodeRecoveryPhase(replaceNodePlan, index));
+    }
+
+    private Phase getNodeRecoveryPhase(Plan inputPlan, int index) {
+        // Get IP address for replacement node.
+        Optional<Protos.TaskStatus> statusOptional = stateStore.fetchStatus(String.format("node-%d-server", index));
+
+        if (!statusOptional.isPresent()) {
+            logger.error("Task {} scheduled for recovery, but doesn't exist");
+            return null;
+        }
+
+        Protos.TaskStatus status = statusOptional.get();
+        String replaceIp = status.getContainerStatus()
+                .getNetworkInfosList().get(0)
+                .getIpAddressesList().get(0)
+                .getIpAddress();
+
+        Phase inputPhase = inputPlan.getChildren().get(0);
+        Step inputLaunchStep = inputPhase.getChildren().get(index);
+
+        // Dig all the way down into the command, so we can append the replace_address option to it.
+        PodInstance podInstance = inputLaunchStep.start().get().getPodInstance();
+        PodSpec podSpec = podInstance.getPod();
+        TaskSpec taskSpec = podSpec.getTasks().stream().filter(t -> t.getName().equals("server")).findFirst().get();
+        CommandSpec command = taskSpec.getCommand().get();
+
+        DefaultCommandSpec.Builder builder = DefaultCommandSpec.newBuilder(command);
+        builder.value(String.format(
+                "%s -Dcassandra.replace_address=%s -Dcassandra.consistent.rangemovement=false%n",
```

Comment:
Do you know if `-Dcassandra.consistent.rangemovement=false` needs to be turned back on at some point ?